stonesoup/models/measurement/nonlinear.py

```diff
@@ -264,8 +264,11 @@ def function(self, state, noise=False, **kwargs) -> StateVector:
 
         # Convert to Spherical
         rho, phi, theta = cart2sphere(*xyz_rot)
+        elevations = [Elevation(i) for i in np.atleast_1d(theta)]
+        bearings = [Bearing(i) for i in np.atleast_1d(phi)]
+        rhos = np.atleast_1d(rho)
 
-        return StateVector([[Elevation(theta)], [Bearing(phi)], [rho]]) + noise
+        return StateVectors([elevations, bearings, rhos]) + noise
```

Comment:
Just incase I'm confusing myself here, the StateVectors type will hstack these to have elevation, bearing, rho rows?

Reply:
That's right. So the 0 index gives another StateVectors object with all the elevations, 1 gives bearings, and 2 gives rhos.